File: app/src/main/java/com/example/david/lists/view/itemlist/buldlogic/ItemListViewComponent.java
package com.example.david.lists.view.itemlist.buldlogic;

import android.app.Application;

import com.example.david.lists.common.buildlogic.ViewCommonModule;
import com.example.david.lists.common.buildlogic.ViewScope;
import com.example.david.lists.view.common.TouchHelperCallback;
import com.example.david.lists.view.itemlist.IItemViewContract;
import com.example.david.lists.view.itemlist.ItemListView;

import dagger.BindsInstance;
import dagger.Component;

@ViewScope
@Component(modules = {
        ItemListViewModule.class,
        ViewCommonModule.class
})
public interface ItemListViewComponent {
    void inject(ItemListView view);

    @Component.Builder
    interface Builder {
        ItemListViewComponent build();

        @BindsInstance
        Builder application(Application application);

        @BindsInstance
        Builder view(IItemViewContract.View view);

        @BindsInstance
        Builder movementCallback(TouchHelperCallback.MovementCallback movementCallback);

        @BindsInstance
        Builder userListId(String userListId);
    }
}

File: app/src/main/java/com/example/david/lists/view/itemlist/buldlogic/ItemListViewComponent.kt
package com.example.david.lists.view.itemlist.buldlogic

import android.app.Application

import com.example.david.lists.common.buildlogic.ViewCommonModule
import com.example.david.lists.common.buildlogic.ViewScope
import com.example.david.lists.view.common.TouchHelperCallback
import com.example.david.lists.view.itemlist.IItemViewContract
import com.example.david.lists.view.itemlist.ItemListView

import dagger.BindsInstance
import dagger.Component

@ViewScope
@Component(modules = [
    ItemListViewModule::class,
    ViewCommonModule::class
])
interface ItemListViewComponent {
    fun inject(view: ItemListView)

    @Component.Builder
    interface Builder {
        fun build(): ItemListViewComponent

        @BindsInstance
        fun application(application: Application): Builder

        @BindsInstance
        fun view(view: IItemViewContract.View): Builder

        @BindsInstance
        fun movementCallback(movementCallback: TouchHelperCallback.MovementCallback): Builder

        @BindsInstance
        fun userListId(userListId: String): Builder
    }
}
